src/Native/System.Security.Cryptography.Native/pal_x509.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_x509.h"
+
+extern "C" X509_NAME* DecodeX509Name(const unsigned char* buf, int32_t len)
```

Comment:
X509Certificates will end up using the methods that are in X509Name as you make further shimming progress; and the SafeHandle should probably move into Interop.X509Name.cs as well (or have the file further exploded).

So my preferences would be
1. Move this to pal_x509_name, and Interop.X509Name.cs
2. Move this to pal_x509_name, move the extern to Interop.X509Name_Decode.cs or Interop.DecodeX509Name.cs
3. Move this to pal_x509_name, leave it in Interop.X509.cs
4. Leave the native method in pal_x509, the extern method in Interop.X509.cs
5. Leave the native method in pal_x509, move the extern to Interop.X509Name.cs